x-pack/plugins/search_indices/public/hooks/api/use_index.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Index } from '@kbn/index-management';
+import { useQuery } from '@tanstack/react-query';
+import { useKibana } from '../use_kibana';
+
+
+export const useIndex = (indexName: string) => {
+  const { http } = useKibana().services;
+  const queryKey = ['fetchIndex', indexName];
+  const result = useQuery({
+    queryKey,
+    queryFn: () => http.fetch<Index>(`/internal/index_management/indices/${encodeURIComponent(indexName)}`),
```

Comment:
this is good for now, but we will most likely be updating this to a `search_indices` route so that we can customize the response.